test/test_table_helper.hpp

```diff
@@ -0,0 +1,179 @@
+/*************************************************************************
+ *
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **************************************************************************/
+
+#ifndef TEST_TEST_TABLE_HELPER_HPP_
+#define TEST_TEST_TABLE_HELPER_HPP_
+
+namespace {
+
+enum Days { Mon, Tue, Wed, Thu, Fri, Sat, Sun };
+
+class TestTable : public realm::Table {
```

Comment:
Clever idea :)

Even though it was intended to add those templated setters on Table as a side effect of the PR; but we can add those in a separate PR later (probably a good idea to keep them separated anyway)